### DISCUSSION TITLE
Thunderbird 102.4: Provider login dialog disappears immediately, no login possible

### DISCUSSION THREAD

**Discussion by CarloGraziani** [kewisch/gdata-provider#519] [Support]
Hi.
I just installed Thunderbird 102.4 from a tarball on Ubuntu 20.04. When I installed the Provider for Google Calendar, all seemed to go well right up to the point when I attempted to connect to my calendar. Then a window -- which I think is almost certainly the Google login window -- blinks in and out, leaving behind a "Create New Calendar" window with a blank list of calendars and tasks to choose from.

I'm not sure how to troubleshoot this. Is there a debug log that I can turn on?

**Comment by harrischrisa** [comment]
I am also running Thunderbird 102.4 on windows 10 and just installed Provider for Google Calendar 102.0.0. I have the same problem. The sign into your account dialog box pops up monetarily and then no calendars listed. On the computer I am logged into my Google account and have the Google Calendar displayed on my browser.

Any suggestions to what I have to do?
Thanks

**Reply by gundown64** [reply]
I have the same issue as you. Also running 102.4 on Windows 10.

**Comment by carbonbaseunit** [comment]
Ditto.

**Comment by schnittchen** [comment]
This started happening yesterday, for me on Linux.

**Reply by abnqeut** [reply]
Same here, provider version 102.0.0 on Thunderbird 102.3.3 on Ubuntu 22.10.

**Comment by SiriusResearch** [comment]
Same here - Windows 10 Pro - Thunderbird 102.4.1 
I'll paste the text from the error console, but basically it's saying: You can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure.

**Reply by SiriusResearch** [reply]
Calendar: [calGoogleSession] Authentication failure: {"error":"\n\u000finvalid_request\u0012Þ\u0001\nYou can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure.\n\nYou can let the app developer know that this app doesn't comply with one or more Google validation rules.\n  \u001a3https://support.google.com/accounts/answer/12379384 �\u0003*.\n\fredirect_uri\u0012\u001eurn:ietf:wg:oauth:2.0:oob:auto2£\u0002\b\u0001\u0012Í\u0001The out-of-band (OOB) flow has been blocked in order to keep users secure. Follow the Out-of-Band (OOB) flow Migration Guide linked in the developer docs below to migrate your app to an alternative method.\u001aOhttps://developers.google.com/identity/protocols/oauth2/resources/oob-migration","details":"\n\u000finvalid_request\u0012Þ\u0001\nYou can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure.\n\nYou can let the app developer know that this app doesn't comply with one or more Google validation rules.\n  \u001a3https://support.google.com/accounts/answer/12379384 �\u0003*.\n\fredirect_uri\u0012\u001eurn:ietf:wg:oauth:2.0:oob:auto2£\u0002\b\u0001\u0012Í\u0001The out-of-band (OOB) flow has been blocked in order to keep users secure. Follow the Out-of-Band (OOB) flow Migration Guide linked in the developer docs below to migrate your app to an alternative method.\u001aOhttps://developers.google.com/identity/protocols/oauth2/resources/oob-migration"} gdataSession.jsm:309

[Exception... "
�invalid_request���
You can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure.

You can let the app developer know that this app doesn't comply with one or more Google validation rules.
  �3https://support.google.com/accounts/answer/12379384 ��*.
�redirect_uri��urn:ietf:wg:oauth:2.0:oob:auto2�������The out-of-band (OOB) flow has been blocked in order to keep users secure. Follow the Out-of-Band (OOB) flow Migration Guide linked in the developer docs below to migrate your app to an alternative method.�Ohttps://developers.google.com/identity/protocols/oauth2/resources/oob-migration"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: resource://gdata-provider/legacy/modules/gdataSession.jsm :: login/authFailed< :: line 311"  data: no] 2 gdataSession.jsm:311:25
    authFailed resource://gdata-provider/legacy/modules/gdataSession.jsm:311
    <anonymous> self-hosted:1115
    onPromptAuthAvailable resource://gdata-provider/legacy/modules/gdataSession.jsm:330
    onAuthorizationFailed resource://gdata-provider/legacy/modules/OAuth2.jsm:226
    _checkForRedirect resource://gdata-provider/legacy/modules/OAuth2.jsm:145
    onLocationChange resource://gdata-provider/legacy/modules/OAuth2.jsm:187


Unknown localization message errors.
�invalid_request���
you can't sign in to this app because it doesn't comply with google's oauth 2.0 policy for keeping apps secure.

you can let the app developer know that this app doesn't comply with one or more google validation rules.
  �3https://support.google.com/accounts/answer/12379384 ��*.
�redirect_uri��urn:ietf:wg:oauth:2.0:oob:auto2�������the out-of-band (oob) flow has been blocked in order to keep users secure. follow the out-of-band (oob) flow migration guide linked in the developer docs below to migrate your app to an alternative method.�ohttps://developers.google.com/identity/protocols/oauth2/resources/oob-migration ExtensionCommon.jsm:2070
    localizeMessage resource://gre/modules/ExtensionCommon.jsm:2070
    localizeMessage resource://gre/modules/Extension.jsm:1542
    getMessage chrome://extensions/content/parent/ext-i18n.js:19
    gdataCalendarsShow resource://gdata-provider/legacy/modules/ui/gdata-calendar-creation.jsm:320

**Comment by MisterDeeDee** [comment]
same problem here.

**Comment by january3** [comment]
Same here, TB 102.4.1, provider version 102.0.0, Ubuntu 20.04

**Comment by tpellard** [comment]
I have the same problem with Thunderbird 102.2.2 under Linux Mint 20.2

**Comment by GileanEU** [comment]
Same problemn here. Never used this plugin before and installed it today, so I thought there was something wrong with my google account or something. "Good" to know I am not the only one then...

**Comment by Frenchwolfie** [comment]
Same problem with W10 22H2 and Thunderbird 102.4.1 64 bits

**Comment by brad** [comment]
I think we have enough reports, right? Probably newcomers here can just click the subscribe button to the right to stay informed.

**Comment by DeeCeeArch** [comment]
Same issue on all machines that have had to re-log in

**Comment by woody0130** [comment]
Same issue for me.  I decided to try and unsubscribe, remove and reinstall and now I can't even create the calendar again.

**Comment by nelsonpoon** [comment]
Same issue for me. Oh no....web calendar for work around = ="

**Reply by harrischrisa** [reply]
Obviously this is a widespread problem with many reports on a variety of platforms.
Is it likely that this issue will be resolved?

**Reply by nelsonpoon** [reply]
Yes, but I can't found out the problem, the screen disappear very fast

**Reply by nelsonpoon** [reply]
> Obviously this is a widespread problem with many reports on a variety of platforms. Is it likely that this issue will be resolved?

Hi, I have turn on the debug , but where to check the Tools> error console ?

**Reply by nelsonpoon** [reply]
reference to [SiriusResearch](https://github.com/SiriusResearch) Comment, it can't solve by users.

**Comment by nelsonpoon** [comment]
Dear All, 

For you information, 

![image](https://user-images.githubusercontent.com/1738644/199675557-66982c67-4f4e-417b-a91d-3991f64e15fd.png)

Calendar: [calGoogleSession] Authentication failure: {"error":"\n\u000finvalid_request\u0012Þ\u0001\nYou can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure.\n\nYou can let the app developer know that this app doesn't comply with one or more Google validation rules.\n  \u001a3https://support.google.com/accounts/answer/12379384 �\u0003*.\n\fredirect_uri\u0012\u001eurn:ietf:wg:oauth:2.0:oob:auto2£\u0002\b\u0001\u0012Í\u0001The out-of-band (OOB) flow has been blocked in order to keep users secure. Follow the Out-of-Band (OOB) flow migration guide linked in the developer docs below to migrate your app to an alternative method.\u001aOhttps://developers.google.com/identity/protocols/oauth2/resources/oob-migration","details":"\n\u000finvalid_request\u0012Þ\u0001\nYou can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure.\n\nYou can let the app developer know that this app doesn't comply with one or more Google validation rules.\n  \u001a3https://support.google.com/accounts/answer/12379384 �\u0003*.\n\fredirect_uri\u0012\u001eurn:ietf:wg:oauth:2.0:oob:auto2£\u0002\b\u0001\u0012Í\u0001The out-of-band (OOB) flow has been blocked in order to keep users secure. Follow the Out-of-Band (OOB) flow migration guide linked in the developer docs below to migrate your app to an alternative method.\u001aOhttps://developers.google.com/identity/protocols/oauth2/resources/oob-migration"}


[Exception... "
�invalid_request���
You can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure.

You can let the app developer know that this app doesn't comply with one or more Google validation rules.
  �3https://support.google.com/accounts/answer/12379384 ��*.
�redirect_uri��urn:ietf:wg:oauth:2.0:oob:auto2�������The out-of-band (OOB) flow has been blocked in order to keep users secure. Follow the Out-of-Band (OOB) flow migration guide linked in the developer docs below to migrate your app to an alternative method.�Ohttps://developers.google.com/identity/protocols/oauth2/resources/oob-migration"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: resource://gdata-provider/legacy/modules/gdataSession.jsm :: login/authFailed< :: line 311"  data: no] 2 [gdataSession.jsm:311:25](resource://gdata-provider/legacy/modules/gdataSession.jsm)
    authFailed resource://gdata-provider/legacy/modules/gdataSession.jsm:311
    <anonymous> self-hosted:1115
    onPromptAuthAvailable resource://gdata-provider/legacy/modules/gdataSession.jsm:330
    onAuthorizationFailed resource://gdata-provider/legacy/modules/OAuth2.jsm:226
    _checkForRedirect resource://gdata-provider/legacy/modules/OAuth2.jsm:145
    onLocationChange resource://gdata-provider/legacy/modules/OAuth2.jsm:187

**Comment by glda19** [comment]
i installed tb 91.13.1 with Provider voor Google Agenda 91.3
And that is the error that i've got
Error 400 invalid_request
And by details i saw this

If you're the app developer, make sure this request information complies with Google's policies.
redirect_uri: urn:ietf:wg:oauth:2.0:oob:auto

hope this help to solve the problem

**Comment by xjonx1** [comment]
The app has to be updated. The developer has been contacted. The problem is the app needs googles new OAuth 2.0.

**Comment by Jokacola** [comment]
Let's hope the developer sorts this soon. I have same problem, provider login box flashes momentarily, disappearing before one has time to react.
Win11, TB 102.4.2

**Comment by scuthber** [comment]
I have the same version of Thunderbird on two PCs - my old windows 10, which works, my new windows 11, which doesn't...

**Comment by Qurtosis** [comment]
> Hi. I just installed Thunderbird 102.4 from a tarball on Ubuntu 20.04. When I installed the Provider for Google Calendar, all seemed to go well right up to the point when I attempted to connect to my calendar. Then a window -- which I think is almost certainly the Google login window -- blinks in and out, leaving behind a "Create New Calendar" window with a blank list of calendars and tasks to choose from.
> 

As many others, I use Thunderbird 102.4.2 and noticed the same behavior, when I attempted to integrate my google calendar. 

As a workaround, I configured Thunderbird as an email client for gmail (which I am not interested in). Thunderbird will properly invoke the google authentication procedure. Once this is done, the google calendars are recognized and the synchronisation works.

**Reply by glda19** [reply]
hi have the same. so as every one.
But i can use the google callendar. Without the google provider addon.
I the left column where you see the callendars i click right en make a new callendar on network and there i fill in my google mail address and then he search for the callendars. I can add new appointment in tb delete

**Reply by CJL48** [reply]
Okay, what if we don't have a gmail account but do have a google account? What site address do we enter then?

**Reply by bagraham** [reply]
@CJL48 Just enter the email address associated with the google account/calendar and leave the location blank.

**Reply by xjonx1** [reply]
@CJL48 All Google users/account holders have a Gmail account, along with a YouTube channel, Google+(or at least they did) and Google docs/drive account, among others.

You don't have to use them but signing up automatically adds them to your profile.

**Comment by jollymover** [comment]
@glda19 Woohoo! It works like a charm!

**Comment by glda19** [comment]
@CJL48 I think when YouTube make a google account its create a Gmail adsress

**Comment by glda19** [comment]
Can some one help me to add the tasks

**Comment by bagraham** [comment]
Thank you @glda19 I and I suspect several others were totally over thinking this.
All you do is hit + to add a calendar.
Then select "On the Network"
Then in the box labeled "Username" enter the email account associated with the google account you want the calendar for (in my case that is not a gmail address)
Leave Location blank and hit "Find Calendars".
Thunderbird goes off and auto discovers the calendar and pops up the google sign in window, login (even with 2FA) and it works a dream.
I've tried this on Windows 10 and 11 and it worked both times. The Windows 10 machine was already connected to that email address the Windows 11 was not.

**Reply by xjonx1** [reply]
I can confirm that this "work around" does work **HOWEVER,** the snooze/dismiss does not. I tried it on three different accounts.

I was able to download, add and delete calendar items, but items that had reminders turned on kept reappearing after snoozing them and the Google OAUth 2.0 window still appears and disappears. I didn't check the log but I suspect that there is still an error for OAuth like before.

Here's hoping that Phil gets on this very soon.

**Comment by nitwoo** [comment]
I had the same issue as everyone else, couldn't login with the prompt window disappearing immediately.

However : the whole point of using the provider is to also have your Google tasks sync in Thunderbird. Which with this method it does not, at least not for me. Looked everywhere, forced syncing, etc. Nothing.

What you are describing is actually just Thunderbird's built-in CalDAV support. The Provider add-on adds a third option when adding a calendar, named "Google Calendar", not local or network. You can totally remove the add-on and do the exact same thing, I literally just tried it. For people who want full read-access sync with tasks, it still requires the provider, which is bugged as a lot of people already reported.

I wouldn't call that a proper solution just yet.

**Reply by CJL48** [reply]
Same. I can get a link to load the calendar into thunderbird, but it will be read only. This includeds being unable to snooze/dismiss reminders.

**Reply by xjonx1** [reply]
I never said it was the solution. I said it was a "work around" until Phil can fix it. 

Use the bodge or not; it's your choice.

**Comment by Jokacola** [comment]
Well, strangely, mine has started working again - both yesterday and today (14/11/22 and 15/11/22). Provider login box appears and stays, allowing me to log in and synch my calendar. Has anyone else had this?

**Comment by nneul** [comment]
Author released a fixed version 102.0.3 already, try updating in add-ons screen.

**Reply by nneul** [reply]
Note that it does NOT appear to fix the hanging calendar sync issue so far, but I'm not 100% certain of that yet.

**Comment by xjonx1** [comment]
I can confirm that my problem is fixed (Google OAuth 2 window self-closing and calendar not updating).

I have tried and set up three different calendars on two different Windows machines.

Process:

1. Download plug-in from [https://addons.thunderbird.net/en-us/thunderbird/addon/provider-for-google-calendar/](https://addons.thunderbird.net/en-us/thunderbird/addon/provider-for-google-calendar/)
2. Open the Thunderbird calendar.
3. Install the plug in that was downloaded.
     3.1 Go to add-ons and themes.
     3.2 Click on the gear icon.
     3.3 Choose "Install Add-on from File"
     3.4 Point add-on window to the file that was downloaded.
4. Be sure to open the viewing menu for calendars.
     4.1 Click on Calendar at right to open the calendar interface.
     4.2 Click on View/Calendar/Calendar Pane and check "Show Calendar Pane" if it is not already displayed.
5. Remove your old Google calendar by right clicking on it and choosing "unsubscribe"
6.  Click the "+" next to Calendar at the top to add a new calendar.
7.  Choose Google Calendar.
8.  Enter the Gmail account associated with the calendar you want to use.
9.  Log into Google. You will get a security email that you can ignore. I have chosen to "Stay singed in" but it is your choice.
10. Choose the Google Calendar you wish to synchronize. Must choose at least one.
11. Repeat for other accounts you want to sync. Right click on a calendar name to change its color or other properties.
12. Uncheck the "Show Calendar Pane" (step 4 - 4.2, above) to hide the sidebar. (optional)

**Comment by rudimatz** [comment]
Fixed for me, thanks a lot! I didn't have to re-login to google as previous comment mentioned under 9)

**Reply by xjonx1** [reply]
I did on one account, so I added it.😁

**Comment by Tom54497** [comment]
Unfortunately sync still is not reliable with that new version. Have to close and start Thunderbird to get the latest appointments.

**Comment by harrischrisa** [comment]
The extension did enable me to subscribe the Google Calendar. I also do have problems with events not synchronising; I don't know if this is the extension or Lightning.

**Comment by glda19** [comment]
If you only use the calender of Google and not the task
You don't need the extension.
You can delete the extension and open the calendar.
Delete all the calendars
Richt click in the left column and then add calendar
Select network and use your Gmail address and there is a working calendar

**Reply by harrischrisa** [reply]
If you subscribe by this method and not the provider, one can NOT use email reminders.

**Comment by madtom1999** [comment]
Still not working for me on Xubuntu LTS 2022.  I can add entries on TB that dissapear when I change them to gmail calendars then appear on the google calendars but the gmail calendars and tasks do not come the other way.

**Reply by xjonx1** [reply]
Did you take a look at your logs? That's how I found the OAuth 2.0 problem.

**Reply by madtom1999** [reply]
I've successfully logged into the accounts via the OAuth 2.0 window and have been offered the calendars and task lists to select.

**Reply by gnartS** [reply]
If its any use what worked for me was (i) create a GMAIL imap account in Thunderbird and log in (this does the OAuth 2 part).
Then add the google calendar with the gdata-provider.

Ive got full read/write and sync access with that method.

**Comment by madtom1999** [comment]
On 01/12/2022 18:34, gnartS wrote:
>
> If its any use what worked for me was (i) create a GMAIL imap account 
> in Thunderbird and log in (this does the OAuth 2 part).
> Then add the google calendar with the gdata-provider.
>
> Ive got full read/write and sync access with that method.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/kewisch/gdata-provider/discussions/519#discussioncomment-4285986>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ABJUUC3VXWZTGA4ZTEFPUY3WLDVUZANCNFSM6AAAAAAROKJEBA>.
> You are receiving this because you commented.Message ID: 
> ***@***.***>
>
I have access to the imap accounts. I seem to have access to the 
calendars - I can add a calendar event on the thunderbird addon for 
google calendar that appears on my phone and the google calender on the 
browser but doesn show on the laptop where I put it in!!!!!!

Tom

**Reply by madtom1999** [reply]
Oops turns out its a corrupted sqlite.cache file.
